src/main/java/nextstep/fp/StreamStudy.java

```diff
@@ -1,33 +1,38 @@
 package nextstep.fp;
 
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class StreamStudy {
 
     public static long countWords() throws IOException {
-        String contents = new String(Files.readAllBytes(Paths
-                .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
+        String contents = Files.readString(Paths.get("src/main/resources/fp/war-and-peace.txt"));
         List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
 
-        long count = 0;
-        for (String w : words) {
-            if (w.length() > 12) count++;
-        }
-        return count;
+        return words.stream()
+                .filter(word -> word.length() > 12)
+                .count();
     }
 
     public static void printLongestWordTop100() throws IOException {
-        String contents = new String(Files.readAllBytes(Paths
-                .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
+        String contents = Files.readString(Paths.get("src/main/resources/fp/war-and-peace.txt"));
         List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
 
-        // TODO 이 부분에 구현한다.
+        printWords(words);
+    }
+
+    private static void printWords(List<String> words) {
+        words.stream()
+                .filter(word -> word.length() > 12)
+                .sorted(Comparator.comparingInt(String::length).reversed())
+                .distinct()
+                .limit(100)
+                .forEach(word -> System.out.print(word.toLowerCase()));
```

Comment:
map 을 함께 사용하며 println 으로 해주어야 개행이 될 것 같아요 🙂

```suggestion
                .map(String::toLowerCase)
                .forEach(System.out::println);
```